DataProcessing/Program.cs

```diff
@@ -0,0 +1,419 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Configuration;
+using QuantConnect.DataSource;
+using QuantConnect.Logging;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using QuantConnect.Securities.Future;
+using QuantConnect.Util;
+
+namespace QuantConnect.DataProcessing
+{
+    public class Program
+    {
+        private const string _futuresBaseUrl = "https://www.cboe.com/us/futures/market_statistics/historical_data/products/csv/";
+        private const string _userAgent = "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0";
+        private const string _csvHeader = "Date,First Month,F1,F2,F3,F4,F5,F6,F7,F8,F9,F10,F11,F12,Contango 2/1,Contango 7/4,Con 7/4 div 3";
+
+        private static readonly DateTime _utcDate = DateTime.UtcNow.Date;
+        private static readonly RateGate _rateGate = new RateGate(1, TimeSpan.FromSeconds(5));
+        
+        public static void Main()
+        {
+            var tempOutputDirectory = Directory.CreateDirectory(Config.Get("temp-output-directory", "/temp-output-directory"));
+            var processedDataFolder = Directory.CreateDirectory(Config.Get("processed-data-directory", Globals.DataFolder));
+            
+            var overwriteExistingEntries = Config.GetBool("overwrite-existing-entries");
+            var outputVendorDirectory = Config.Get("output-vendor-directory", "vixcentral");
+            
+            var startDateValue = Environment.GetEnvironmentVariable("QC_DATAFLEET_DEPLOYMENT_DATE");
+            var startDate = startDateValue != null
+                ? Parse.DateTimeExact(startDateValue, "yyyyMMdd", DateTimeStyles.None)
+                : _utcDate;
+
+            var vxFutures = GetFutureChain(startDate);
+            var vxCloses = DownloadFuturesSettlement(vxFutures);
+
+            var contango = GetContango(vxCloses, vxFutures);
```

Comment:
For a given date we are returning a list of values here, would of expected a single data point for that date? Not sure we want the other values?

Reply:
I'll gate this behind a config variable. It's useful so that if we ever have to reprocess the data from scratch we can do it in a single run, but yes I totally agree :+1: